test/new_relic/agent/agent_test.rb

```diff
@@ -475,7 +475,7 @@ def test_wait_raises_if_not_connected_once_signaled
         @agent.signal_connected
         thread.join
 
-        assert error, error
+        assert_equal error, error
```

Comment:
These puzzle me a bit. Why wouldn't `error == error`? Why might we need this assertion?

Reply:
My hunch is that `assert` required truthiness before, whereas `assert_equal` does not. In the test, `error` starts out `nil` and is expected to become non-nil by the time the assertions take place.

@hannahramadan I think you can satisfy both my hunch and Kayla's concern with this change:

```diff
- assert_equal error, error
+ refute_nil error
```

Reply:
Granted `nil` will also fail the `kind_of` check the next assertion performs immediately afterwards. So if you'd prefer, you can get rid of the `error, error` assertion entirely instead of changing it.

Reply:
I was also confused on this, but assumed I was missing something about the code 😅 I'll remove the assert entirely.